packages/lodestar/src/chain/attestation/pool.ts

```diff
@@ -106,7 +106,10 @@ export class AttestationProcessor implements IAttestationProcessor {
   public async receiveBlock(signedBlock: SignedBeaconBlock): Promise<void> {
     // process block's attestations
     const attestations = signedBlock.message.body.attestations;
-    await Promise.all(Array.from(attestations).map((a) => this.receiveAttestation(a)));
+    // process one by one in order to cache checkpoit state
+    for (const attestation of attestations) {
+      await this.receiveAttestation(attestation);
+    }
```

Comment:
we already ordered pending attestations based on a block root already at below code